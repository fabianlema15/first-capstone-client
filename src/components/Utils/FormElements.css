labela {
  display: block;
}

.fieldset label{
  text-align: left;
  width: 38%;
}

.fieldset input, .fieldset  button, .fieldset  select, .fieldset  textarea{
  text-align: left;
  width: 58%;
}

.fieldset  button {
  text-align: center;
  height: 26px;
  padding: 0;
  margin: 0;
}
