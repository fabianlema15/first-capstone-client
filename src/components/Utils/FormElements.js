import React, { Component } from 'react';
import './FormElements.css'

function Input({ className, label, idx, id, ...props }) {
  return (
    <div className='fieldset'>
      <label htmlFor={id}>
        {label}
      </label>
      <input id={id} required className={['Input', className].join(' ')} {...props} />
    </div>
  )
}

function Textarea({ className, label, idx, id, ...props }) {
  return (
    <div className='fieldset'>
      <label key={idx} htmlFor={id}>
        {label}
      </label>
      <textarea id={id} className={['Textarea', className].join(' ')} {...props} />
    </div>
  )
}

function Select({ className, label, idx, id, options=[], ...props }) {
  const getOptions = options.map(option => <option key={option.id} value={option.id}>{option.value}</option>)
  return (
    <div className='fieldset'>
      <label key={idx} htmlFor={id}>
        {label}
      </label>
      <select id={id} required className={['Textarea', className].join(' ')} {...props}>
        <option value=''>--SELECT ONE OPTION--</option>
        {getOptions}
      </select>
    </div>
  )
}

class File extends Component{
  ref = React.createRef();

  onClickButton() {
    this.ref.current.click();
  }

  render(){
    const { className, label, idx, id, buttonTitle, ...props } = this.props;
    return (
      <div className='fieldset'>
        <label key={idx} htmlFor={id}>
          {label}
          <input id={id} type="file" ref={this.ref} required accept=".png,.jpg,.jpeg" className={['PictureButton', className].join(' ')} {...props} />

        </label>
        <button type='button' className='green' onClick={() => this.onClickButton()}>{buttonTitle || 'Select Image'}</button>
      </div>
    )
  }
}

/*function File({ className, label, idx, id, buttonTitle, pictureAction, ...props }) {
  return (
    <fieldset>
    <label key={idx} htmlFor={id}>
      {label}
      <input id={id} type="file" ref={ref} required accept=".png,.jpg,.jpeg" className={['PictureButton', className].join(' ')} {...props} />
      <button type='button' className='green' onClick={pictureAction}>{buttonTitle || 'Select Image'}</button>
    </label>
    </fieldset>
  )
}*/

const ColoredLine = ({ color }) => (
    <hr
        style={{
            color: color,
            backgroundColor: color,
            height: 3
        }}
    />
);

export { Input, Textarea, Select, File, ColoredLine };
