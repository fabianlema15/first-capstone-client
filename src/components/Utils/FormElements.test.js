import { Input, Textarea, Select, File, ColoredLine } from './FormElements'
import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<FormElements />', () => {
  it('renders Input without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Input/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders Input as expected', () => {
	  const wrapper = shallow(<Input/>);
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

    it('renders Textarea without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Textarea/>, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Renders Textarea as expected', () => {
  	  const wrapper = shallow(<Textarea/>);
  	  expect(toJson(wrapper)).toMatchSnapshot();
  	  });

      it('renders Select without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
          <Select/>, div);
        ReactDOM.unmountComponentAtNode(div);
      });

      it('Renders Select as expected', () => {
    	  const wrapper = shallow(<Select/>);
    	  expect(toJson(wrapper)).toMatchSnapshot();
    	  });

        it('renders File without crashing', () => {
          const div = document.createElement('div');
          ReactDOM.render(
            <File/>, div);
          ReactDOM.unmountComponentAtNode(div);
        });

        it('Renders File as expected', () => {
      	  const wrapper = shallow(<File/>);
      	  expect(toJson(wrapper)).toMatchSnapshot();
      	  });

          it('renders ColoredLine without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(
              <ColoredLine/>, div);
            ReactDOM.unmountComponentAtNode(div);
          });

          it('Renders ColoredLine as expected', () => {
        	  const wrapper = shallow(<ColoredLine/>);
        	  expect(toJson(wrapper)).toMatchSnapshot();
        	  });

})
