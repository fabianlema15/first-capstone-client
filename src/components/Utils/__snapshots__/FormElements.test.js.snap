// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormElements /> Renders ColoredLine as expected 1`] = `
<hr
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
      "height": 3,
    }
  }
/>
`;

exports[`<FormElements /> Renders File as expected 1`] = `
<div
  className="fieldset"
>
  <label>
    <input
      accept=".png,.jpg,.jpeg"
      className="PictureButton "
      required={true}
      type="file"
    />
  </label>
  <button
    className="green"
    onClick={[Function]}
    type="button"
  >
    Select Image
  </button>
</div>
`;

exports[`<FormElements /> Renders Input as expected 1`] = `
<div
  className="fieldset"
>
  <label />
  <input
    className="Input "
    required={true}
  />
</div>
`;

exports[`<FormElements /> Renders Select as expected 1`] = `
<div
  className="fieldset"
>
  <label />
  <select
    className="Textarea "
    required={true}
  >
    <option
      value=""
    >
      --SELECT ONE OPTION--
    </option>
  </select>
</div>
`;

exports[`<FormElements /> Renders Textarea as expected 1`] = `
<div
  className="fieldset"
>
  <label />
  <textarea
    className="Textarea "
  />
</div>
`;
